core/build/cli.odin

```diff
@@ -0,0 +1,182 @@
+package build
+
+import "core:strings"
+import "core:os"
+import "core:path/filepath"
+import "core:sync"
+import "core:encoding/json"
+import "core:runtime"
+import "core:fmt"
+import "core:slice"
+import "core:unicode/utf8"
+import "core:unicode"
+import "core:strconv"
+import "core:intrinsics"
+
+
+Flag_Arg :: struct { // -flag:key=val
+	flag: string,
+	key: string,
+	val: string,
+}
```

Comment:
One issue with Odin's flag argument style is that it doesn't really allow for pipe-line passing, or general shell expansion. 

I chose this approach in Odin purely because it was as simple and consistent, even if it has its issues. It's also the reason why I have never written a flag library for Odin because which approach do you choose?--given that each approach has its flaws.

Reply:
I don't think you can "choose" an approach for a flag library if it needs to be in `core`, as every approach is so fundamentally different from each other (unix style allowing multiple single letter flags under a single `-` (`-rf` or `-r -f`). 

My thoughts on that is that you can provide styles (`.Odin .Unix .Windows`), and do different things based on the style of flags the app supports. For example, `core:build` would use the `core:argument_parser` or whatever with the `.Odin` style.

Reply:
I'd have some ideas for this in order to make the "setup" cross-style compatible, i might give it a shot eventually when i get some extra time